lynxkite-app/web/src/workspace/NodeSearch.tsx

```diff
@@ -12,82 +17,388 @@ export type OpsOp = {
+          }}
+        >
+          <span style={{ display: "flex", alignItems: "center", gap: "0.5rem" }}>
+            <ArrowLeftIcon style={{ width: "14px", height: "14px" }} />
+            Back
+          </span>
+        </div>
+      )}
+
+      <div style={styles.list}>
+        {/* CHANGED: Using map instead of forEach for better performance */}
+        {results.map((result, index) => (
           <div
-            key={box.item.name}
-            tabIndex={0}
-            onFocus={() => setSelectedIndex(index)}
-            onMouseEnter={() => setSelectedIndex(index)}
-            onClick={addSelected}
-            className={`search-result ${index === selectedIndex ? "selected" : ""}`}
+            key={result.category ? `${result.category}-${result.item}` : result.item}
+            className={`node-search-item ${result.isCategory ? "node-search-item-category" : ""} ${index === selectedIndex ? "node-search-item-selected" : ""}`}
+            /* NEW: Using onMouseDown instead of onClick to prevent blur/select race condition */
+            onMouseDown={(e) => {
+              e.preventDefault();
+              if (result.op) {
+                handleItemClick(result.op);
+              } else if (result.isCategory) {
+                handleCategoryClick(result.item);
+              }
+            }}
+            onMouseEnter={() => {
+              setSelectedIndex(index);
+            }}
           >
-            {box.item.categories.map((category) => `${category}\u00A0›\u00A0`)}
-            {box.item.name}
+            {/* NEW: Added icons for better visual distinction */}
+            {result.isCategory ? (
+              <span style={{ display: "flex", alignItems: "center", gap: "0.5rem" }}>
+                <FolderIcon style={{ width: "16px", height: "16px", color: "#007bff" }} />
+                {result.item}
+              </span>
+            ) : result.category ? (
+              <span>
+                {result.item}{" "}
+                <span style={{ color: "#666", fontSize: "0.8em" }}>({result.category})</span>
+              </span>
+            ) : (
+              result.item
+            )}
           </div>
         ))}
       </div>
     </div>
   );
 }
+
+// NEW: Using CSS-in-JS object for better maintainability
+const styles: Record<string, React.CSSProperties> = {
+  container: {
+    position: "absolute",
+    border: "1px solid #aaa",
+    width: "300px",
+    maxHeight: "400px",
+    overflowY: "auto",
+    overflowX: "hidden",
+    padding: "8px",
+    background: "#fff",
+    fontFamily: "sans-serif",
+    borderRadius: "6px",
+    zIndex: 1000,
+    resize: "none",
+  },
+  search: {
+    marginBottom: "8px",
+    width: "100%",
+    padding: "6px",
+    boxSizing: "border-box",
+    borderRadius: "4px",
+    border: "1px solid #ccc",
+  },
+  // NEW: Added styled back button for better navigation UX
+  backButton: {
+    marginBottom: "10px",
+    cursor: "pointer",
+    color: "#007bff",
```

Comment:
Let's not introduce new colors either. I suggest you try to work with the existing rules in `index.css`, modifying them and expanding them as needed.